base/namedtuple.jl

```diff
@@ -113,6 +113,15 @@ function convert(::Type{NamedTuple{names,T}}, nt::NamedTuple{names}) where {name
     NamedTuple{names,T}(T(nt))
 end
 
+function Tuple(nt::NamedTuple{names}) where {names}
+    if @generated
+        return Expr(:tuple, Any[:(getfield(nt, $(QuoteNode(n)))) for n in names]...)
+    else
+        return tuple(nt...)
+    end
+end
+(::Type{T})(nt::NamedTuple{names, T}) where {names, T <: Tuple} = Tuple(nt)
```

Comment:
This should return a tuple of type `T`.

Reply:
Unless I misunderstand something, this seems guaranteed by the method above (note the `T` here matches the storage inside the `NamedTuple`).

I should probably also made have a method that lest you `convert` the values and so-on, but for now that remains slow (I just didn't want this particular method to be a performance trap, since pulling `T` directly from the `NamedTuple` and using that rather than `Tuple` seems like a reasonable thing for a user to do).

Reply:
I think I can remove that restriction and make the right hand side `convert(T, Tuple(nt))` (and that will deal with all the `Union` and `UnionAll` cases that I didn't want to bother with here).

Reply:
Fixed.

Reply:
You're right, I missed the second occurrence of the `T`.